### PR TITLE
[Draft] Dart sdk

### PR DESCRIPTION
**Pull request by lohanidamodar**
Hey,
- I have created Dart SDK templates,
- cleaned up Dart language file a bit with some fixes
- Removed Flutter specific and unused dependencies

Running into incorrect parameter type issue in functions api as mentioned here #76

### CODE REVIEW

**Comment by TorstenDittmann** [comment]
@lohanidamodar Uhh nice 👀 

Will this also be compatible with Flutter Web? I remember it being a problem in the past.

**Comment by lohanidamodar** [comment]
@TorstenDittmann we have separate flutter SDK which is already compatible for flutter web as well

**Comment by TorstenDittmann** [comment]
> @TorstenDittmann we have separate flutter SDK which is already compatible for flutter web as well

Ah, so Dart SDK != Flutter SDK? 😄 
That confused me for a moment 👍

File: templates/dart/lib/client.dart.twig [hunk 1]
```diff
@@ -76,17 +59,10 @@ class Client {
 
     Future init() async {
         if(!initialized) {
-          final Directory cookieDir = await _getCookiePath();
-
-          cookieJar = new PersistCookieJar(dir:cookieDir.path);
-
+          cookieJar = new PersistCookieJar(dir:'./cookies');
```
Comment (eldadfux) [inline templates/dart/lib/client.dart.twig:62]: I think we don't need the cookie jar for the server integration. No need for any cookie based auth when integrating with the server API, we'll only use API Key & JWT for authentication.

**Comment by eldadfux** [comment]
> > @TorstenDittmann we have separate flutter SDK which is already compatible for flutter web as well
> 
> Ah, so Dart SDK != Flutter SDK? 😄
> That confused me for a moment 👍

Yeh, our naming conventions for SDKs are sdk-for-[platform]. On the server the platform name will be the language name (sdk-for-php, sdk-for-python). On the client it would be the platform name (sdk-for-web, sdk-for-flutter, sdk-for-ios, sdk-for-android, sdk-for-unity...). That's also the reason for changing sdk-for-js to sdk-for-web

File: templates/dart/lib/client.dart.twig (hunk 1 above)
Comment (lohanidamodar) [reply templates/dart/lib/client.dart.twig:62]: @eldadfux Ok

**Pull request merged**